cognite/neat/core/_client/data_classes/statistics.py

```diff
@@ -0,0 +1,125 @@
+
+@dataclass
+class InstanceCountsLimits(InstanceCounts):
+    instances_limit: int
+    soft_deleted_instances_limit: int
+
+    @property
+    def instances(self) -> int:
+        return self.nodes + self.edges
+
+    @property
+    def soft_deleted_instances(self) -> int:
+        return self.soft_deleted_nodes + self.soft_deleted_edges
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            nodes=resource["nodes"],
+            edges=resource["edges"],
+            soft_deleted_nodes=resource["softDeletedNodes"],
+            soft_deleted_edges=resource["softDeletedEdges"],
+            instances_limit=resource["instancesLimit"],
+            soft_deleted_instances_limit=resource["softDeletedInstancesLimit"],
+        )
+
+
+@dataclass
+class ProjectStatsAndLimits(CogniteResource):
+    project: str
+    spaces: CountLimitPair
+    containers: CountLimitPair
+    views: CountLimitPair
+    data_models: CountLimitPair
+    container_properties: CountLimitPair
+    instances: InstanceCountsLimits
+    concurrent_read_limit: int
+    concurrent_write_limit: int
+    concurrent_delete_limit: int
+
+    @classmethod
+    def _load(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            project=data["project"],
+            spaces=CountLimitPair._load(data["spaces"]),
+            containers=CountLimitPair._load(data["containers"]),
+            views=CountLimitPair._load(data["views"]),
+            data_models=CountLimitPair._load(data["dataModels"]),
+            container_properties=CountLimitPair._load(data["containerProperties"]),
+            instances=InstanceCountsLimits._load(data["instances"]),
+            concurrent_read_limit=data["concurrentReadLimit"],
+            concurrent_write_limit=data["concurrentWriteLimit"],
+            concurrent_delete_limit=data["concurrentDeleteLimit"],
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return {
+            "project": self.project,
+            "spaces": self.spaces.dump(camel_case=camel_case),
+            "containers": self.containers.dump(camel_case=camel_case),
+            "views": self.views.dump(camel_case=camel_case),
+            "dataModels" if camel_case else "data_models": self.data_models.dump(camel_case=camel_case),
+            "containerProperties" if camel_case else "container_properties": self.container_properties.dump(
+                camel_case=camel_case
+            ),
+            "instances": self.instances.dump(camel_case=camel_case),
+            "concurrentReadLimit" if camel_case else "concurrent_read_limit": self.concurrent_read_limit,
+            "concurrentWriteLimit" if camel_case else "concurrent_write_limit": self.concurrent_write_limit,
+            "concurrentDeleteLimit" if camel_case else "concurrent_delete_limit": self.concurrent_delete_limit,
+        }
```

Comment:
Q: Why does this type implement `dump` itself instead of using the inherited implementations from `CogniteResource`?

Reply:
The default `dump` method you get from `CogniteResource` does not handle nested object. We do the same in the PySDK.